File: src/main/java/org/jukeboxmc/item/behavior/ItemBanner.java
package org.jukeboxmc.item.behavior;

import org.jukeboxmc.block.Block;
import org.jukeboxmc.block.BlockType;
import org.jukeboxmc.item.Burnable;
import org.jukeboxmc.item.Item;
import org.jukeboxmc.item.ItemType;
import org.jukeboxmc.util.Identifier;

import java.time.Duration;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class ItemBanner extends Item implements Burnable {

    public ItemBanner( Identifier identifier ) {
        super( identifier );
    }

    public ItemBanner( ItemType itemType ) {
        super( itemType );
    }

    @Override
    public Block toBlock() {
        return Block.create( BlockType.STANDING_BANNER );
    }

    @Override
    public Duration getBurnTime() {
        return Duration.ofMillis( 300 );
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/item/behavior/ItemBanner.kt
package org.jukeboxmc.server.item.behavior

import org.jukeboxmc.api.block.Block
import org.jukeboxmc.api.block.BlockType
import org.jukeboxmc.api.item.Burnable
import org.jukeboxmc.api.item.ItemType
import org.jukeboxmc.server.item.JukeboxItem
import java.time.Duration

class ItemBanner(itemType: ItemType, countNetworkId: Boolean) : JukeboxItem(itemType, countNetworkId), Burnable {

    override fun getBurnTime(): Duration {
        return Duration.ofMillis(300)
    }

    override fun toBlock(): Block {
        return Block.create(BlockType.STANDING_BANNER)
    }

}